src/mongoc/mongoc-init.c

```diff
@@ -90,6 +80,8 @@ static MONGOC_ONCE_FUN( _mongoc_do_init)
 #endif
 
 #ifdef MONGOC_ENABLE_SASL
+   /* The following functions should not use tracing, as they may be invoked
+    * before mongoc_log_set_handler() can complete. */
```

Comment:
@ajdavis: this seemed preferable to copy/pasting it for each of the four functions.